config.go

```diff
@@ -55,14 +84,29 @@ func LoadConfig(configFile string) (*menderConfig, error) {
 		return nil, err
 	}
 
+	if confFromFile.Servers != nil {
+		// Overwrite "active" server with first one from the list of servers.
+		confFromFile.ServerURL = confFromFile.Servers[0].ServerURL
+		confFromFile.TenantToken = confFromFile.Servers[0].TenantToken
```

Comment:
So `Servers` essentially replaces `ServerURL` in the configuration file? If so, I would produce a hard error here if both are present, just to be sure there is no doubt about which server(s) gets used.

Same goes for the `TenantToken`.

Reply:
Following this, I would consider only using the `Servers` list internally. IOW, instead of first trying `ServerURL` and then resorting to the list, just use the list in either case, starting with the first element. This might also simplify some of the fallback logic.

If only `ServerURL` is specified in the config file, you can just add this to the list as the only element.

Reply:
The idea is that the whole `Servers` entry is optional, and the active one used is "loaded" into the `ServerURL`, `TenantToken` entry in the base structure. This preserves backward compatibility. Note that when the `Servers` field is specified, the `ServerURL` and `TenantToken` fields are "don't care" fields when parsing the configuration file. If `Servers` is not specified, then the client behaves as before.

Reply:
I'm not sure if I want to replace base fields completely with the list structure, as I don't see how this simplifies the logic. Maybe I'm missing something?